src/NuGet.Clients/NuGet.PackageManagement.PowerShellCmdlets/NuGetPowerShellBaseCommand.cs

```diff
@@ -1095,7 +1099,21 @@ public void ReportError(string message)
 
         public NuGetActionType ActionType { get; set; }
 
-        public INuGetTelemetryService TelemetryService { get; set; }
+        public KeyValuePair<string, Guid> OperationId
+        {
+            get
+            {
+                if (_operationId.Value == Guid.Empty)
+                {
+                    _operationId = new KeyValuePair<string, Guid>(nameof(OperationId), Guid.NewGuid());
+                }
+                return _operationId;
```

Comment:
I see this in a couple spots, I suggest adding a helper method to check and init this.